plasmapy/tests/helpers/tests/test_expected.py

```diff
@@ -0,0 +1,178 @@
+        pytest.fail(
+            f"The ExpectedTestOutcome instance for {case.argument} did not "
+            f"raise the expected exception but instead returned {result}."
+        )
+
+
+is_warning_test_inputs = [
+    (Warning, True),
+    (UserWarning, True),
+    (PlasmaPyWarning, True),
+    (Exception, False),
+    (BaseException, False),
+    ("", False),
+    ((Warning, 1), False),
+]
+
+
+@pytest.mark.parametrize("possible_warning, actually_a_warning", is_warning_test_inputs)
+def test__is_warning(possible_warning, actually_a_warning: bool):
+    """
+    Test that `~plasmapy.utils.pytest_helpers.expected._is_warning`
+    returns `True` for warnings and `False` for other objects.
+    """
+    interpreted_as_warning = _is_warning(possible_warning)
+    if interpreted_as_warning != actually_a_warning:
+        pytest.fail(
+            f"_is_warning({repr(possible_warning)}) should return "
+            f"{actually_a_warning}, but is instead returning "
+            f"{interpreted_as_warning}."
+        )
+
+
+is_exception_test_inputs = [
+    (Warning, False),
+    (UserWarning, False),
+    (Exception, True),
+    (BaseException, True),
+    ("", False),
+]
+
+
+@pytest.mark.parametrize(
+    "possible_exception, actually_an_exception", is_exception_test_inputs
+)
+def test_is_exception(possible_exception, actually_an_exception: bool):
+    """
+    Test that `~plasmapy.utils.pytest_helpers.expected._is_exception`
+    returns `True` for exceptions and `False` for other objects.
+    """
+    interpreted_as_exception = _is_exception(possible_exception)
+    if interpreted_as_exception != actually_an_exception:
+        pytest.fail(
+            f"_is_exception({repr(possible_exception)} should return "
+            f"{actually_an_exception}, but is instead returning "
+            f"{interpreted_as_exception}."
+        )
+
+
+is_warning_and_value_test_inputs = [
+    ((Warning, ""), True),
+    (["", UserWarning], True),
+    ((Warning, UserWarning), False),
+    (Warning, False),
+    (UserWarning, False),
+    (Exception, False),
+    (BaseException, False),
+    ("", False),
+]
+
+
+@pytest.mark.parametrize(
+    "possible_warning_and_value, actually_warning_and_value",
+    is_warning_and_value_test_inputs,
+)
+def test__is_warning_and_value(
+    possible_warning_and_value, actually_warning_and_value: bool
+):
+    """
+    Test that `_is_warning_and_value` returns `True` for a `tuple` or
+    `list` containing a warning and an `object` that is not a `Warning`,
+    and `False` for anything else.
+    """
+    interpreted_as_warning_and_value = _is_warning_and_value(possible_warning_and_value)
+    if interpreted_as_warning_and_value != actually_warning_and_value:
+        pytest.fail(
+            f"_is_warning_and_value returns {interpreted_as_warning_and_value} "
+            f"with {repr(possible_warning_and_value)} as an argument, but is "
+            f"expected to return {actually_warning_and_value}."
+        )
```

Comment:
```suggestion
    assert interpreted_as_warning_and_value == actually_warning_and_value
```
Seriously, make these fail on your local setup and see what gets printed out!